src/api/firebase.js

```diff
@@ -91,3 +91,21 @@ export async function createNewList(listId) {
 	}
 	return response;
 }
+
+export async function checkIfListExists(listId) {
+	const listCollectionRef = collection(db, listId);
+
+	try {
+		const querySnapshot = await getDocs(listCollectionRef);
+
+		// If the collection exists and has documents, return the snapshot
+		if (!querySnapshot.empty) {
```

Comment:
**Praise:** Nice work figuring out the `getDocs` function from Firestore and using the `.empty` check!